datasets/um005/um005.py

```diff
@@ -0,0 +1,137 @@
+
+
+_DESCRIPTION = """\
+UMC005 English-Urdu is a parallel corpus of texts in English and Urdu language with sentence alignments. The corpus can be used for experiments with statistical machine translation.
+
+The texts come from four different sources:
+- Quran
+- Bible
+- Penn Treebank (Wall Street Journal)
+- Emille corpus
+
+We provide the religious texts of Quran and Bible for direct download. Because of licensing reasons, Penn and Emille texts cannot be redistributed freely. However, if you already hold a license for the original corpora, we are able to provide scripts that will recreate our data on your disk. Our modifications include but are not limited to the following:
+
+- Correction of Urdu translations and manual sentence alignment of the Emille texts.
+- Manually corrected sentence alignment of the other corpora.
+- Our data split (training-development-test) so that our published experiments can be reproduced.
+- Tokenization (optional, but needed to reproduce our experiments).
+- Normalization (optional) of e.g. European vs. Urdu numerals, European vs. Urdu punctuation, removal of Urdu diacritics.
+"""
+_HOMEPAGE_URL = "http://ufal.ms.mff.cuni.cz/umc/005-en-ur/"
+_URL = "http://ufal.ms.mff.cuni.cz/umc/005-en-ur/download.php?f=umc005-corpus.zip"
+_CITATION = """\
+@unpublished{JaZeWordOrderIssues2011,
+author      = {Bushra Jawaid and Daniel Zeman},
+title       = {Word-Order Issues in {English}-to-{Urdu} Statistical Machine Translation},
+year        = {2011},
+journal     = {The Prague Bulletin of Mathematical Linguistics},
+number      = {95},
+institution = {Univerzita Karlova},
+address     = {Praha, Czechia},
+issn        = {0032-6585},
+}
+"""
+
+_ALL = "all"
+_VERSION = "1.0.0"
+_SOURCES = ["bible", "quran"]
+_SOURCES_FILEPATHS = {
+    s: {
+        "train": {"urdu": "train.ur", "english": "train.en"},
+        "dev": {"urdu": "dev.ur", "english": "dev.en"},
+        "test": {"urdu": "test.ur", "english": "test.en"},
+    }
+    for s in _SOURCES
+}
+
+
+class UM005Config(datasets.BuilderConfig):
+    def __init__(self, *args, sources=None, **kwargs):
+        super().__init__(*args, version=datasets.Version(_VERSION, ""), **kwargs)
+        self.sources = sources
+
+    @property
+    def filepaths(self):
+        return {source: _SOURCES_FILEPATHS[source] for source in self.sources}
+
+
+class UM005(datasets.GeneratorBasedBuilder):
+    BUILDER_CONFIGS = [
+        UM005Config(name=source, sources=[source], description=f"Source. {source}.") for source in _SOURCES
+    ] + [UM005Config(name=_ALL, sources=_SOURCES, description=f"All sources included.")]
+
+    DEFAULT_CONFIG_NAME = _ALL
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "urdu": datasets.Value("string"),
+                    "english": datasets.Value("string"),
```

Comment:
Could you use the `Translation` feature type ?
see [flores](https://github.com/huggingface/datasets/blob/master/datasets/flores/flores.py) for an example of usage

Reply:
what do you think ?

Reply:
Will fix at some point today. Got distracted by more interesting datasets :)